macro/src/main/scala/monocle/macros/Focus.scala

```diff
@@ -0,0 +1,35 @@
+package monocle.macros
+
+import scala.language.dynamics
+
+import monocle._
+import monocle.syntax.ApplyLens
+
+/** Syntactic sugar around an ApplyLens */
+class Focus[S,A](val applyLens: ApplyLens[S,S,A,A]) extends AnyVal with Dynamic{
+  def apply[C](field: A => C): Focus[S,C] = macro internal.FocusImpl.apply[S,A,C]
+
+  /**
+  Whitebox macro for field access syntax.
+  E.g. someobj.focus.a.b...
+  */
+  def selectDynamic(field: String): Any = macro internal.FocusImpl.selectDynamicImpl
+  
+  /**
+  Assignment syntax for updates
+  E.g. someobj.focus(_.a.b) = ...
+  */
+  def update[C](field: A => C, value: C): S = macro internal.FocusImpl.update[S,A,C]
+  /**
+  Assignment syntax for updates with member access syntax
+  E.g. someobj.focus.a.b = ...
+  */
+  def updateDynamic[C](field: String)(value: C): S = macro internal.FocusImpl.updateDynamicImpl[S,C]
```

Comment:
Just to be clear, the whitebox macro and Dynamic are both not needed for the assignment syntax. They are only needed for the `.focus.a.b` field syntax (selectDynamic for building the lens, updateDynamic for assignment). If you use `.focus(_.a.b)` you need neither and get code completion in IntelliJ.

I agree to be careful with these features, but since we have a syntactically more verbose, but better supported "fallback", I think it is ok in this case. It means we are providing slightly more risky, fully optional syntactic sugar, which we can deprecate if we have to. Also, having worked closely with @odersky and @xeno-by for a few years, I doubt they are going away any time soon if ever.